lib/engine/game/g_1822_mrs/game.rb

```diff
@@ -0,0 +1,688 @@
+            price: 200,
+            rusts_on: '6',
+          },
+          {
+            name: '4',
+            distance: 4,
+            num: 4,
+            price: 300,
+            rusts_on: '7',
+          },
+          {
+            name: '5',
+            distance: 5,
+            num: 2,
+            price: 500,
+            events: [
+              {
+                'type' => 'close_concessions',
+              },
+            ],
+          },
+          {
+            name: '6',
+            distance: 6,
+            num: 3,
+            price: 600,
+            events: [
+              {
+                'type' => 'full_capitalisation',
+              },
+            ],
+          },
+          {
+            name: '7',
+            distance: 7,
+            num: 20,
+            price: 750,
+            variants: [
+              {
+                name: 'E',
+                distance: 99,
+                multiplier: 2,
+                price: 1000,
+              },
+            ],
+            events: [
+              {
+                'type' => 'phase_revenue',
+              },
+            ],
+          },
+          {
+            name: '2P',
+            distance: 2,
+            num: 2,
+            price: 0,
+          },
+          {
+            name: 'LP',
+            distance: [
+              {
+                'nodes' => ['city'],
+                'pay' => 1,
+                'visit' => 1,
+              },
+              {
+                'nodes' => ['town'],
+                'pay' => 1,
+                'visit' => 1,
+              },
+            ],
+            num: 1,
+            price: 0,
+          },
+          {
+            name: '5P',
+            distance: 5,
+            num: 1,
+            price: 0,
+          },
+          {
+            name: 'P+',
+            distance: [
+              {
+                'nodes' => ['city'],
+                'pay' => 99,
+                'visit' => 99,
+              },
+              {
+                'nodes' => ['town'],
+                'pay' => 99,
+                'visit' => 99,
+              },
+            ],
+            num: 2,
+            price: 0,
+          },
+        ].freeze
+
+        UPGRADE_COST_L_TO_2_PHASE_2 = 70
+
+        def discountable_trains_for(corporation)
+          discount_info = []
+
+          upgrade_cost = if @phase.name.to_i < 2
+                           self.class::UPGRADE_COST_L_TO_2
+                         else
+                           self.class::UPGRADE_COST_L_TO_2_PHASE_2
+                         end
+          corporation.trains.select { |t| t.name == 'L' }.each do |train|
+            discount_info << [train, train, '2', upgrade_cost]
+          end
+          discount_info
+        end
+
+        def init_companies(players)
+          # Make sure we have the correct starting companies
+          self.class.const_set(:STARTING_COMPANIES, self.class::STARTING_COMPANIES_ADVANCED) if optional_advanced?
```

Comment:
don't use const set, just turn this into an instance method with an override

Reply:
Ok will do